app/src/main/java/com/mwalasiak/pyramid/LevelsActivity.kt

```diff
@@ -143,31 +129,44 @@ class LevelsActivity : AppCompatActivity() {
             }
 
             return copyOfLevel.toString().replace(" ", "").removeSurrounding("[", "]")
-        }else
+        }else {
             return EMPTY_LIST_TEXT
+        }
 
     }
 
-    private fun setSharedPreferences(): SharedPreferences {
-        return getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE)
-    }
-
     private fun verifySharedPref(key: String): Boolean {
         val sharedValue = setSharedPreferences()
         val prefKeyName = sharedValue.getString(key, null)
         return prefKeyName != null
     }
 
+    private fun getList(listName: String): String? {
+        return setSharedPreferences().getString(listName, null)
+    }
+
+    private fun setList(stringWithNumbers: String?, listKey: String) {
+        val editor = setSharedPreferences().edit()
+        editor.putString(listKey, stringWithNumbers)
+        editor.apply()
+    }
+
+    private fun getTextView(id: Int) = binding.root.findViewById<TextView>(id)
+
+    private fun setSharedPreferences(): SharedPreferences {
+        return getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE)
+    }
+
     private fun setToastMessage() {
         val emptySaveNumbersInput = Toast.makeText(applicationContext, NO_NUMBERS_INPUT, Toast.LENGTH_SHORT)
         emptySaveNumbersInput.show()
     }
 
-    private fun getNumbersFromLottoService() {
+    /*private fun getNumbersFromLottoService() {
         viewModelScope.launch{
             val numbers = LottoApi.retrofitService.getLatestMultiMultiNumbers()
         }
-    }
+    }*/
 
     private fun EditText.clear() {text.clear()}
```

Comment:
I guess I know why you've added it, but this "logic" is to use to shortcut it :D Good you are learning about extension functions :+1: good to know you know them already, but I don't think it's worth having this one ;)